ros_buildfarm/workspace.py

```diff
@@ -99,6 +99,8 @@ def call_build_tool(
         setup_file = os.path.join(parent_result_space, 'setup.sh')
         if os.path.exists(setup_file):
             cmd = '. %s && %s' % (setup_file, cmd)
+            if build_tool == 'colcon':
+                cmd = 'COLCON_CURRENT_PREFIX="%s" %s' % (parent_result_space, cmd)
```

Comment:
This looks like a hack. Can you explain why that is done?

Reply:
This would be necessary if the underlay workspace wasn't located in the exact same path in the container at which it was built. For most packages, setting `COLCON_CURRENT_PREFIX` is enough to let `setup.sh` know about the change.

I thought that this was sufficient until I tried building things on top of the rviz packages, which proved to have hard-coded paths making them non-relocatable. So I ended up re-structuring this job so that the underlay can sit at the same path, even though it shouldn't really be necessary.

So this can be removed now that we're limited to a single underlay and it sits in the same spot, but if any of that changes, this would be necessary since `setup.sh` can't detect its own location reliably, right?

Reply:
> This would be necessary if the underlay workspace wasn't located in the exact same path in the container at which it was built. For most packages, setting `COLCON_CURRENT_PREFIX` is enough to let `setup.sh` know about the change.

Ok, I understand now why you used it. So setting it to make the `.sh` relocatable sounds fine.

> I thought that this was sufficient until I tried building things on top of the rviz packages, which proved to have hard-coded paths making them non-relocatable. So I ended up re-structuring this job so that the underlay can sit at the same path, even though it shouldn't really be necessary.

Not sure if that is a good idea. The same problem would appear if users would download the underlay archive and unpack it elsewhere. Maybe it would actually be good to keep using a different location to actually catch that problem?

Reply:
Agreed. I opened https://github.com/ros2/ros2/issues/606 to discuss the issue, and it sounds non-trivial. For now, I'd rather keep the workaround in place than render overlay jobs useless for a significant number of packages.

Reply:
:+1: